app/services/work_packages/set_schedule_service.rb

```diff
@@ -144,9 +144,11 @@ def reschedule_by_predecessors(scheduled, dependency)
   def reschedule_to_date(scheduled, date)
     new_start_date = [scheduled.start_date, date].compact.max
 
+    duration = scheduled.duration || 0;
+
     set_dates(scheduled,
               new_start_date,
-              scheduled.due_date && (new_start_date + scheduled.duration - 1))
+              scheduled.due_date && (new_start_date + duration - 1))
```

Comment:
The alternative would probably be to change this line to the following:

```
scheduled.due_date && !scheduled.duration.nil? && (new_start_date + duration - 1)
```

I'm not sure which is best or more correct.

Reply:
This version would be better, since duration should not ever be 0, it's either nil or bigger than 0.

Reply:
This version would be better, since duration should not ever be 0, it's either nil or bigger than 0.

Reply:
Thanks, changed